src/world.cpp

```diff
@@ -94,8 +92,17 @@ World::update_single_mesh(TerrainDim3 tile_sop) {
 
 void
 World::update_all_chunks_mesh() {
-    initialize_chunks_mesh_();
-    for (size_t i = 0; i < chunks_mesh_.size(); i++)
+    size_t num_chunks = terrain_main_.get_chunks().size();
+    if (chunks_mesh_.size() != num_chunks) {
+        chunks_mesh_.reserve(num_chunks);
+        for (size_t i = 0; i < num_chunks; i++) {
+            chunks_mesh_.push_back(std::make_shared<gui::data_structures::TerrainMesh>(
+                                       gui::data_structures::TerrainMesh()
+            ));
+        }
+    }
```

Comment:
When will this `if` statement run? How many times will it run?

Reply:
Not very often. I had it running twice. I think it runs once now.

Reply:
but i mean

will `num_chunks` ever decrease?

Reply:
If num chunks isn't constant size then chunks would be a map or something. This would be dynamic loading of chunks.

Reply:
point is, should you be doing `!=` or `<`

Also, shouldn't you `clear()`?